src/core/annotation.js

```diff
@@ -877,17 +888,315 @@ class WidgetAnnotation extends Annotation {
     return !!(this.data.fieldFlags & flag);
   }
 
-  getOperatorList(evaluator, task, renderForms) {
+  getOperatorList(evaluator, task, renderForms, annotationStorage) {
     // Do not render form elements on the canvas when interactive forms are
     // enabled. The display layer is responsible for rendering them instead.
     if (renderForms) {
       return Promise.resolve(new OperatorList());
     }
-    return super.getOperatorList(evaluator, task, renderForms);
+    return super.getOperatorList(
+      evaluator,
+      task,
+      renderForms,
+      annotationStorage
+    );
+  }
+}
+
+class WidgetWithTextAnnotation extends WidgetAnnotation {
```

Comment:
The classes in this file directly match the specification which makes it useful to map it one-to-one. I would therefore not introduce a new base class here, mainly because it adds yet another level of inheritance which makes it harder to follow, but also because this is not a class that the specification contains. Instead, we can simply put these methods on the `WidgetAnnotation` class itself and overwrite them in subclasses if necessary. This way we have one less level of inheritance while still keeping the functionality.